tokio-trace-fmt/src/default.rs

```diff
@@ -10,44 +14,163 @@ use std::fmt::{self, Write};
 #[cfg(feature = "ansi")]
 use ansi_term::{Colour, Style};
 
-pub fn fmt_event<N>(ctx: &span::Context<N>, f: &mut Write, event: &Event) -> fmt::Result
+/// A type that can measure and format the current time.
+///
+/// This trait is used by `Standard` to include a timestamp with each `Event` when it is logged.
+///
+/// Notable default implementations of this trait are `SystemTime` and `()`. The former prints the
+/// current time as reported by `std::time::SystemTime`, and the latter does not print the current
+/// time at all. `FormatTime` is also automatically implemented for any function pointer with the
+/// appropriate signature.
+pub trait FormatTime {
+    /// Measure and write out the current time.
+    ///
+    /// When `format_time` is called, implementors should get the current time using their desired
+    /// mechanism, and write it out to the given `fmt::Write`. Implementors must insert a trailing
+    /// space themselves if they wish to separate the time from subsequent log message text.
+    fn format_time(&self, w: &mut fmt::Write) -> fmt::Result;
+}
+
+impl FormatTime for () {
+    fn format_time(&self, _: &mut fmt::Write) -> fmt::Result {
+        Ok(())
+    }
+}
+
+impl FormatTime for fn(&mut fmt::Write) -> fmt::Result {
+    fn format_time(&self, w: &mut fmt::Write) -> fmt::Result {
+        (*self)(w)
+    }
+}
+
+/// Retrieve and print the current wall-clock time.
+///
+/// If the `chrono` feature is enabled, the current time is printed in a human-readable format like
+/// "Jun 25 14:27:12.955". Otherwise the `Debug` implementation of `std::time::SystemTime` is used.
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Default)]
+pub struct SystemTime;
```

Comment:
It seems like we might also want to have an implementation that prints _uptime_, rather than timestamps. This is the approach we use in `linkerd2-proxy`, so that we can use `Instant::now` rather than `SystemTime::now` and avoid a syscall per log line on some platforms. The uptime stamps _can_ be turned into wall clock time by adding them to the process' start time, and can be compared to each other in the log.

We may also want to consider having a `tokio_timer::clock::now`-based `FormatTime` implementation eventually